substrate/frame/support/src/traits/tokens/fungibles/lifetime.rs

```diff
@@ -32,6 +32,18 @@ pub trait Create<AccountId>: Inspect<AccountId> {
 	) -> DispatchResult;
 }
 
+/// Trait for refunding the deposit of a target asset account.
+pub trait Refund<AccountId> {
+	/// Means of identifying one asset class from another.
+	type AssetId: AssetId;
+	/// Scalar type for representing balance of an account.
+	type Balance: Balance;
+	/// Returns the amount of account deposit and depositor address, if any.
+	fn deposit_held(id: Self::AssetId, who: AccountId) -> Option<(AccountId, Self::Balance)>;
```

Comment:
I think this function is unnecessary in the `Refund` interface. Having `fn refund` should be enough, the other `fungibles` trait impls can be used to determine the deposits.

Reply:
`fungibles` traits have no function to check the deposit held for an asset's account existence.

Reply:
Here is the example where it is used 
https://github.com/paritytech/polkadot-sdk/pull/3250/files#diff-fbf88fab34bcf066e3c722cd1345365529c1c652df82604ba23aea4f047993d5R243
https://github.com/paritytech/polkadot-sdk/blob/05e639d4bcc3b4d6f99b3ff386155c7dc8115518/substrate/frame/asset-conversion/ops/src/lib.rs#L182

Reply:
I was indeed missing something because that's from another PR :grin: .

I still feel this functionality for querying the deposit shouldn't be in a `Refund` trait, but rather in some other parent interface to separate these concerns. In practical terms though, it doesn't matter too much. In the future, if we need to separate them, it will be very easy to do so.